pkg/security/security_profile/ad.go

```diff
@@ -0,0 +1,602 @@
+func (m *Manager) stopDumpsWithSelector(selector cgroupModel.WorkloadSelector) {
+	counter, ok := m.dumpLimiter.Get(selector)
+	if !ok {
+		counter = atomic.NewUint64(uint64(m.config.RuntimeSecurity.ActivityDumpMaxDumpCountPerWorkload))
+		m.dumpLimiter.Add(selector, counter)
+	} else {
+		if counter.Load() < uint64(m.config.RuntimeSecurity.ActivityDumpMaxDumpCountPerWorkload) {
+			seclog.Infof("activity dumps will no longer be generated for %s", selector.String())
+			counter.Store(uint64(m.config.RuntimeSecurity.ActivityDumpMaxDumpCountPerWorkload))
+		}
+	}
+
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	for _, ad := range m.activeDumps {
+		if adSelector := ad.Profile.GetWorkloadSelector(); adSelector != nil && adSelector.Match(selector) {
+			m.finalizeKernelEventCollection(ad, true)
+			m.dropMaxDumpReached.Inc()
+		}
+	}
+}
+
+// handleSilentWorkloads checks if we should start tracing one of the workloads from a profile without an activity tree of the Security Profile manager
+func (m *Manager) handleSilentWorkloads() {
+	if !m.config.RuntimeSecurity.SecurityProfileEnabled {
+		return
+	}
+
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	// check if it's a good time to look for a silent workload, to do so, check if the last stopped dump was stopped more
+	// than the configured amount of time ago
+	if time.Since(m.lastStoppedDumpTime) < m.config.RuntimeSecurity.ActivityDumpSilentWorkloadsDelay {
+		return
+	}
+
+	// if we're already at capacity leave now - this prevents an unnecessary lock on the security profile manager
+	if len(m.activeDumps) >= m.config.RuntimeSecurity.ActivityDumpTracedCgroupsCount {
+		return
+	}
+
+	// fetch silent workloads
+workloadLoop:
+	for selector, workloads := range m.fetchSilentWorkloads() {
+		if len(workloads) == 0 {
+			// this profile is on its way out, ignore
+			continue
+		}
+
+		if len(m.activeDumps) >= m.config.RuntimeSecurity.ActivityDumpTracedCgroupsCount {
+			// we're at capacity, ignore for now
+			break
+		}
+
+		// check if we already have an activity dump for this selector
+		for _, ad := range m.activeDumps {
+			// the dump selector is resolved if it has been counted by the limiter
+			if !ad.IsCountedByLimiter() {
+				continue
+			}
+
+			adSelector := ad.Profile.GetWorkloadSelector()
+			if adSelector != nil && adSelector.Match(selector) {
+				// we already have an activity dump for this selector, ignore
+				continue workloadLoop
+			}
+		}
+
+		// if we're still here, we can start tracing this workload
+		defaultConfigs, err := m.getDefaultLoadConfigs()
+		if err != nil {
+			seclog.Errorf("couldn't get default load configs: %v", err)
+			continue
+		}
+
+		defaultConfig, found := defaultConfigs[containerutils.CGroupManager(workloads[0].CGroupContext.CGroupFlags)]
+		if !found {
+			seclog.Errorf("Failed to find default activity dump config for %s", containerutils.CGroupManager(workloads[0].CGroupContext.CGroupFlags).String())
```

Comment:
Could we display the CGroupID too ?

Reply:
Done with 1b9ba3974c2b56d0eeed0c41c7117c0166b3f3d9